config/routes.rb

```diff
@@ -7,11 +7,22 @@
 
   # to add a task
   post '/tasks/new' => 'add_task#new'
-  post '/tasks' => 'add_task#create'
+  post '/tasks'     => 'add_task#create'
 
   # to delete a task
   delete '/tasks/delete/:id' => 'delete_task#destroy'
```

Comment:
These routes are not necessarily RESTful - so you'd want to use `/tasks/:id/edit` rather than `tasks/edit/:id` per the default Rails routes.